nifi-nar-bundles/nifi-extension-utils/nifi-event-transport/src/main/java/org/apache/nifi/event/transport/netty/codec/SocketByteArrayMessageDecoder.java

```diff
@@ -38,7 +47,38 @@ public class SocketByteArrayMessageDecoder extends MessageToMessageDecoder<byte[
     protected void decode(final ChannelHandlerContext channelHandlerContext, final byte[] bytes, final List<Object> decoded) {
         final InetSocketAddress remoteAddress = (InetSocketAddress) channelHandlerContext.channel().remoteAddress();
         final String address = remoteAddress.getHostString();
-        final ByteArrayMessage message = new ByteArrayMessage(bytes, address);
+
+        ByteArrayMessage message = getMessageWithSslSessionStatus(channelHandlerContext, bytes, address);
+        if (message == null) {
+            message = new ByteArrayMessage(bytes, address);
+        }
         decoded.add(message);
     }
+
+    private ByteArrayMessage getMessageWithSslSessionStatus(final ChannelHandlerContext channelHandlerContext, final byte[] bytes, final String address) {
+        Iterator<Map.Entry<String, ChannelHandler>> iterator = channelHandlerContext.channel().pipeline().iterator();
+        while (iterator.hasNext()) {
+            final ChannelHandler channelHandler = iterator.next().getValue();
+            if (channelHandler instanceof SslHandler) {
+                return createMessageWithSslSessionStatus((SslHandler)channelHandler, bytes, address);
+            }
+        }
+        return null;
+    }
+
+    private ByteArrayMessage createMessageWithSslSessionStatus(final SslHandler sslHandler, final byte[] bytes, final String address) {
+        final SSLSession sslSession = sslHandler.engine().getSession();
+        try {
+            final Certificate[] certificates = sslSession.getPeerCertificates();
+            if (certificates.length > 0) {
+                final X509Certificate certificate = (X509Certificate) certificates[0];
+                final String subjectDN = certificate.getSubjectDN().toString();
+                final String issuerDN = certificate.getIssuerDN().toString();
```

Comment:
Following the comment to change the `SslSessionStatus` property types, recommend replacing `getSubjectDN()` and `getIssuerDN()` with `getSubjectX500Principal()` and `getIssuerX500Principal()` since the DN methods are denigrated according to the Java 8 documentation:

https://docs.oracle.com/javase/8/docs/api/java/security/cert/X509Certificate.html#getIssuerDN--